src/Microsoft.AspNetCore.SignalR.Core/DefaultHubActivator.cs

```diff
@@ -9,6 +9,8 @@ namespace Microsoft.AspNetCore.SignalR
 {
     public class DefaultHubActivator<THub> : IHubActivator<THub> where THub: Hub
     {
+        // Object factory for all this Hub type
+        private static ObjectFactory _objectFactory = ActivatorUtilities.CreateFactory(typeof(THub), Type.EmptyTypes);
```

Comment:
This is a pretty heavy operation to have in a static constructor. Also if the hub type is registered with the service provider then we'll never even use it. IMO create it when it is needed

Reply:
I looked at that, I can make it lazy.